lib/spack/spack/bootstrap.py

```diff
@@ -278,17 +278,20 @@ def _install_by_hash(self, pkg_hash, pkg_sha256, index, bincache_platform):
             with spack.config.override(
                     'compilers', [{'compiler': compiler_entry}]
             ):
-                spec_str = '/' + pkg_hash
-                parser = argparse.ArgumentParser()
-                spack.cmd.buildcache.setup_parser(parser)
-                install_args = [
-                    'install',
-                    '--sha256', pkg_sha256,
-                    '--only-root',
-                    '-a', '-u', '-o', '-f', spec_str
-                ]
-                args = parser.parse_args(install_args)
-                spack.cmd.buildcache.installtarball(args)
+                # Prevent attempts to bootstrap a compiler when bootstrapping packages
+                # like clingo in case config:install_missing_compilers is set to true
+                with spack.concretize.enable_compiler_existence_check():
```

Comment:
For binaries we install a package hash and the line above this _should_ ensure we have the right compiler available, so I'm surprised this happens. I cannot reproduce the issue in our `spack/ubuntu-bionic` image too. Can you outline a reproducer for the issue?